src/text/non_blocking/text_buffer.rs

```diff
@@ -371,47 +376,53 @@ mod tests {
 
     #[test]
     fn read_from() {
-        let source = r#"first line"#;
+        let source = "first line\n";
         let mut input = TextBuffer::new(source.as_bytes().to_owned());
         let size = input.load_next_line().unwrap();
-        assert_eq!(input.remaining_text().trim(), "first line");
+        assert_eq!(input.remaining_text(), "first line\n");
         input.consume(size);
 
-        let more = "second line";
+        let more = "second line\n";
         match input.read_from(more.as_bytes(), more.len()) {
             Ok(x) if x == more.len() => (),
             wrong => panic!("Unexpected response from read_from: {:?}", wrong),
         }
         input.load_next_line().unwrap();
-        assert_eq!(input.remaining_text().trim(), more);
+        assert_eq!(input.remaining_text(), more);
         assert_eq!(input.lines_loaded(), 2);
     }
 
     #[test]
     fn partial_utf8() {
+        // atom_modified is a multi-byte unicode character with a variation.
+        // The first 3 bytes represent the character '⚛' (atom). The trailing
+        // 3 bytes represent the variation which adds a purple background.
         let atom_modified = &[0xe2, 0x9a, 0x9b, 0xef, 0xb8, 0x8f];
-        let source: &[u8] = &[
-            0x57, 0x65, 0x20, 0x4c, 0x6f, 0x76, 0x65, 0x20, 0x49, 0x6f, 0x6e, 0x21,
-        ];
-        let mut input = TextBuffer::new(source.to_owned());
+        let source: &str = "We Love Ion! ";
+        let mut input = TextBuffer::new(source.as_bytes().to_owned());
+        // Add only part of our atom character, copying all but the last byte of the modifier.
         input
-            .append_bytes(&atom_modified[..5])
+            .append_bytes(&atom_modified[..=4])
             .expect("unable to add partial UTF-8 character");
-        // This should fail since we have a partial UTF-8 sequence in the first line.
+
+        // This should succeed, and provide access to all of the data up to the start of the
+        // incomplete sequence. Unfortunately, the 'invalid' sequence starts with the modifier, so
+        // the atom symbol is returned.
```

Comment:
From the end user perspective this should be ok, right? The text buffer may return the partial glyph ('atom' instead of 'purple background atom') but the Ion reader will say that whatever larger value it's processing (string, symbol, clob) is `Incomplete`.

Reply:
Yes, I believe so. Every string-like needs a container except identifiers, but they are limited to ASCII characters.